bot/bot.py

```diff
@@ -34,16 +35,20 @@ def __init__(
         self.graphql = GraphQLClient(session=session)
 
     async def setup_hook(self) -> None:
-        """Sync the application command tree before bot connects for commands."""
-        await self.tree.sync()
+        """Assign a error handler for Interaction commands."""
+        self.tree.on_error = self.on_command_error
 
-    @staticmethod
-    async def on_ready() -> None:
-        """Runs when the bot is connected."""
+    async def on_ready(self) -> None:
+        """Runs when the bot is connected. Sync Interaction/app_commands when connected to the gateway."""
         log.info('Awaiting...')
         log.info("Bot Is Ready For Commands")
+        await self.tree.sync()
```

Comment:
We should move the `await self.tree.sync()` to above the logging info stating that the bot is ready for commands, or perhaps add a line for tree syncing. `Bot Is Ready For Commands` implies that all commands for the bot are ready and up-to-date, but that does not happen until application commands have finished syncing.

Reply:
Moved above `Bot Is Ready For Commands`